From 09363e538c3f94fe3444a3e09ff52ab0735f1aef Mon Sep 17 00:00:00 2001
From: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Date: Mon, 13 Jan 2020 15:26:34 +0530
Subject: [PATCH 0838/1852] PCI: XDMA PL PCIe: Fixing legacy interrupt handling
 bug

commit 95181323a2a5359b09b51de1d8ff1fa9dd45d6dd from
https://github.com/Xilinx/linux-xlnx.git

Interrupts were tested as per fixes in CR-981433.
Verified interrupts and fixing bug while handling legacy interrupts.

Signed-off-by: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pci/controller/pcie-xdma-pl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/controller/pcie-xdma-pl.c b/drivers/pci/controller/pcie-xdma-pl.c
index bf81c92e11d2..c9f9784a7a80 100644
--- a/drivers/pci/controller/pcie-xdma-pl.c
+++ b/drivers/pci/controller/pcie-xdma-pl.c
@@ -361,8 +361,8 @@ static irqreturn_t xilinx_pcie_intr_handler(int irq, void *data)
 
 	if (status & XILINX_PCIE_INTR_INTX) {
 		/* Handle INTx Interrupt */
-		val =  pcie_read(port, XILINX_PCIE_REG_IDRN);
-		val = val >> XILINX_PCIE_IDRN_SHIFT;
+		intr_val = pcie_read(port, XILINX_PCIE_REG_IDRN);
+		intr_val = intr_val >> XILINX_PCIE_IDRN_SHIFT;
 
 		for_each_set_bit(bit, &intr_val, INTX_NUM)
 			generic_handle_irq(irq_find_mapping(port->leg_domain,
-- 
2.31.1

